src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy

```diff
@@ -39,7 +41,9 @@ class NUnit extends ConventionTask {
 
     File getNunitExec() {
         assert getNunitHome(), "You must install NUnit and set nunit.home property or NUNIT_HOME env variable"
-        File nunitExec = nunitBinFile("nunit-console${useX86 ? '-x86' : ''}.exe")
+        File nunitExec = getNunitVersion().startsWith("3.")
```

Comment:
Yeah, I like how you solved that.
I had a similar thing, but got confused with all the implicitness of gradle/Groovy and finally decided to remove it again.
I might port your approach.